### PR TITLE
add tensorflow max version for keras

### PR DESCRIPTION
**Pull request by forrestwaters**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -629,6 +629,17 @@ def _patch_repodata(repodata, subdir):
                 new_deps.append('python-libarchive-c')
             instructions["packages"][fn]['depends'] = new_deps
 
+        if record['name'] == 'keras':
```
Comment (msarahan) [inline main.py]: Should there be a version upper bound, such that some future version of Keras need not add this constraint?

Comment (msarahan) [reply main.py]: I guess not: https://medium.com/tensorflow/standardizing-on-keras-guidance-on-high-level-apis-in-tensorflow-2-0-bad2b04c819a

Comment (forrestwaters) [reply main.py]: It looks like keras 2.3 added support for tensorflow 2.0. I'll add that as an upper bound for this.

File: main.py [hunk 2]
```diff
@@ -629,7 +629,7 @@ def _patch_repodata(repodata, subdir):
                 new_deps.append('python-libarchive-c')
             instructions["packages"][fn]['depends'] = new_deps
 
-        if record['name'] == 'keras':
+        if record['name'] == 'keras' and record['version'] < '2.3':
```
Comment (msarahan) [inline main.py]: Unfortunately, this version comparison won't work.  At best, it'll do lexicographic sorting, but we need a proper understanding of the version.

**Comment by forrestwaters** [comment]
diff from test-hotfix.py looks good.

**Pull request merged**